.github/workflows/generate.yml

```diff
@@ -0,0 +1,33 @@
+name: Automatic Report and Test Manifest Generation
+on:
+  push:
+    branches: [main]
+    paths: ["reports/**", "tests/**"]
+  pull_request:
+    branches: [ main ]
+    paths: ["reports/**", "tests/**"]
+
+jobs:
+  build:
+    name: Generate Reports using Ruby
+    runs-on: ubuntu-latest
+    permissions:
+      # Give the default GITHUB_TOKEN write permission to commit and push the changed files back to the repository.
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
I suspect you will still need to supply the `GH_TOKEN` here and in the commit action (see https://github.com/eyereasoner/eye-js/blob/a0008130be3389ddf473943dbdcc7c8aaf57943a/.github/workflows/update-eye.yml#L9-L11)

Reply:
The `permissions` key above seemed to do it, as I didn't need it to get it working in my fork, and it did add the automated commit to this PR.

Reply:
LGTM then :)